pdns check 4.0.3 fails to build with gcc 7

Buildlog:

```
g++ -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp         -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -c -o test-dnsname_cc.o test-dnsname_cc.cc
recpacketcache.cc: In member function 'uint64_t RecursorPacketCache::bytes()':
recpacketcache.cc:214:13: error: types may not be defined in a for-range-declaration [-fpermissive]
   for(const struct Entry& e :  d_packetCache) {
             ^~~~~~
make[3]: *** [Makefile:2529: recpacketcache.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

Full buildlog: https://kojipkgs.fedoraproject.org//work/tasks/9023/17769023/build.log

Do you have a patch available?

--- a/pdns/recpacketcache.cc
+++ b/pdns/recpacketcache.cc
@@ -211,7 +211,7 @@ uint64_t RecursorPacketCache::size()
 uint64_t RecursorPacketCache::bytes()
 {
   uint64_t sum=0;
-  for(const struct Entry& e :  d_packetCache) {
+  for(const auto& e :  d_packetCache) {
     sum += sizeof(e) + e.d_packet.length() + 4;
   }
   return sum;

